compiler/AST/foralls.cpp

```diff
@@ -859,77 +900,65 @@ CallExpr* resolveForallHeader(ForallStmt* pfs, SymExpr* origSE)
 //                           //
 ///////////////////////////////
 
-// lowerForallStmts() removes each ForallStmt from AST,
-// replacing it with explicit loop(s) etc.
-// Currently this is done towards the end of resolution.
-// Our goal is to push it to a later pass, ideally all the way
-// to lowerIterators.
-
-void lowerForallStmts() {
-  forv_Vec(ForallStmt, fs, gForallStmts) {
-    if (!fs->inTree() || !fs->getFunction()->isResolved())
-      continue;
+//
+// The block iterRecSetup() is used to fall back to the "old",
+// iterator-record-based implementation for the ForallStmt's
+// parallel iterator. We set it up here so that calls like
+// _getIterator can be resolved.
```

Comment:
Is the iterRecSetup a workaround? Would it be possible to resolve the appropriate bits here and then construct this block during lower iterators?